test/generatortests.js

```diff
@@ -14,7 +15,7 @@ var runGeneratorTests = function(testcases) {
                 // It seems reasonable to expect that the generated xpath
                 // should parse back to the same object, although this may 
                 // not always hold true.
-                equal(parsed.toString(), xpath.parse(parsed.toXPath()).toString(), "" + i + " produced same result when reparsed.");
+                equal(parsed.toString(), xpath.parse(parsed.toHashtag()).toString(), "" + i + " produced same result when reparsed.");
```

Comment:
Changing this test seems like a bad idea. Could we keep this test unmodified, and add a second test that uses `toHashtag()`? (assuming we need to expose `toHashtag` at all, I'm not convinced that we do, more on that elsewhere)